### DISCUSSION TITLE
Disabling auto-login on RaspiOSLite

### DISCUSSION THREAD

**Discussion by eranroll** [gitbls/sdm#29] [Q&A] [answered]
Hi,

I was wondering if there is a way to configure RaspiOSLite image to not auto login with the user I created. This is the command I am running:
```
sudo /usr/local/sdm/sdm --customize --restart --nowpa --locale en_US.UTF-8 --keymap us --user test --password-user test --password-same yes --disable piwiz --svc-disable userconfig  ~/Downloads/2022-04-04-raspios-bullseye-arm64-lite.img
```

I saw a section in the readme about raspi-config automation, but I am not sure how is it done, since I couldn't find anymore documentation. This option does exist in the raspi-config system settings.

**Comment by gitbls** [comment]
Are you sure that autologin is set? I started with a fresh-from-unxz IMG, ran your exact command, burned it to an SD card, and the system came up to the login prompt. I was then able to login using test/test. There was no sign of autologin. 

Can you please do `sudo ls -l /etc/systemd/system/getty@tty1.service.d/` on the running system? I'm curious whether the file autologin.conf is there, and if so, when it was created. It's not in the IMG as distributed by RPT, nor was it in my copy of the IMG after I ran your customize command.

As far as your second question, I'm not sure exactly what you're asking about WRT raspi-config automation. Could you expand on this a bit?

**Comment by gitbls** [comment]
Update: In a flash of post-response brilliance LOL, I reviewed the docs and of course understand your question about raspi-config automation.

You would do raspi-config automation using a Custom Phase script. Copy sdm-customphase to another file (I keep mine in /usr/local/sdm but you can put it anywhere), and in the post-install phase (after the comment "INSERT your Custom Phase post-install code here", you can do
```
   SUDO_USER=${myuser} raspi-config raspi-config-function raspi-config-function-value nonint
```
raspi-config function is the name of the function in raspi-config that you want to call, raspi-config-function-value is the value to pass to that function. "nonint" tells raspi-config to run in "batch" mode: process the command and exit.

For example, to change the boot behavior, to B1 you would do
```
   SUDO_USER=${myuser} raspi-config do_boot_behaviour B1 nonint
```

**Comment by eranroll** [comment]
Hi! So you answered both of my questions :)

I have tried it again and you are correct. Auto login is indeed disabled, so sorry for the trouble. It didn’t work when I omitted the svc-disable part, and I had flashed an older version. So thank you once again.

Regarding the raspi-config, sounds good. I’ll give it a shot.


By the way, where is the cmdline.txt altered? In the post-install code too? In case I wish to disable the console on uart0.

**Comment by gitbls** [comment, marked as answer]
cmdline.txt is modified in sdm-cmdsubs as part of completing the burn command. It a) removes the resize if `--expand-root` was specified (do it at burn time rather than during the first boot), and b) changes the diskid, in the same scenario. Both of these are functions done during the first boot of the system in a "vanilla" (non-sdm updated) system. 

Beyond that, there is no other code in sdm that modifies cmdline.txt. You can verify by: `grep cmdline.txt /usr/local/sdm/*`

If you want to update cmdline.txt, you can do that in your Custom Phase script using sed. The prudent way to formulate these sed commands is to do it interactively on a copy of cmdline.txt.
- Remove something from cmdline.txt: `sed -i "s/ console=tty1//g" /boot/cmdline.txt`, which removes the string "console=tty1" from cmdline.txt
- Add something to cmdline.txt: `sed -i  "s/rootwait/rootwait newthingie//g" /boot/cmdline.txt`, which replaces the string "rootwait" with rootwait and some new text following it.
- Update something in cmdline.txt: `sed -i "s/console=tty1/console=tty2" /boot/cmdline.txt`, which changes console=tty1 to console=tty2
